src/test/java/com/answerdigital/benhession/academy/answerkingweek2/services/OrderServiceTest.java

```diff
@@ -0,0 +1,376 @@
+package com.answerdigital.benhession.academy.answerkingweek2.services;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.answerdigital.benhession.academy.answerkingweek2.exceptions.ConflictException;
+import com.answerdigital.benhession.academy.answerkingweek2.exceptions.ItemUnavailableException;
+import com.answerdigital.benhession.academy.answerkingweek2.exceptions.NotFoundException;
+import com.answerdigital.benhession.academy.answerkingweek2.model.Item;
+import com.answerdigital.benhession.academy.answerkingweek2.model.Order;
+import com.answerdigital.benhession.academy.answerkingweek2.model.OrderItem;
+import com.answerdigital.benhession.academy.answerkingweek2.repositories.OrderRepository;
+
+import java.math.BigDecimal;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(classes = {OrderService.class})
+@ExtendWith(MockitoExtension.class)
+public class OrderServiceTest {
+    @Mock
+    private OrderRepository orderRepository;
+
+    @Mock
+    private ItemService itemService;
+
+    @InjectMocks
+    private OrderService orderService;
+
+    @Test
+    void testAddOrderReturnsSavedOrder() {
+        // Given
+        Order order = new Order("42 Main St");
+
+        // When
+        when(orderRepository.save(any()))
```

Comment:
I think it will be good to narrow down the scope of the orderRepository mock that it accepts to Order class like below: 
when(orderRepository.save(any(Order.class)))
                .thenReturn(order);

Reply:
similarly to the previous comment: 
verify(orderRepository).save(any(Order.class));

Reply:
Should be resolved now!